janitor/utils.py

```diff
@@ -1671,3 +1677,170 @@ def _computations_as_categorical(df: pd.DataFrame, **kwargs) -> pd.DataFrame:
     df = df.astype(categories_dtypes)
 
     return df
+
+
+def _select_columns(
+    df: pd.DataFrame,
+    columns_to_select: Union[str, slice, Pattern, callable, list],
+) -> Union[list, pd.Index]:
+    """
+    Helper function to select columns from a dataframe.
+    The aim is to make this function a base for the various
+    functions in the janitor package that require column selection,
+    while offering a lot of options/flexibility. It is inspired
+    by R's dplyr's select function.
+
+    It accepts a string, slice, callable, regex,
+    a shell-style glob string (e.g., `*_thing_*`), or a list of
+    the previously mentioned options.
+
+    If it is a callable, it is applied to every series in the dataframe
+    and each series must return either True or False, resulting in a
+    sequence of booleans.
+
+    Data types are checked in this function as well.
+
+    A list or pandas Index of columns present in the dataframe is returned.
+    """
+
+    if not isinstance(columns_to_select, list):
+        check(
+            "column label", columns_to_select, [str, slice, Pattern, callable]
+        )
+        return __columns_not_a_list(df, columns_to_select)
+
+    for label in columns_to_select:
+        check("column label", label, [str, slice, Pattern, callable])
+
+    all_columns = []
+    for entry in columns_to_select:
+        outcome = __columns_not_a_list(df, entry)
+        all_columns.extend(outcome)
+
+    if not pd.Series(all_columns).is_unique:  # remove duplicate column names
+        return pd.unique(all_columns)
```

Comment:
I think this check can be simplified. Instead of adding two lines to check, we can check at the `all_columns.extend(outcome)` line. Here's a proposed change:

```python
all_columns.extend([c for c in outcome if c not in all_columns])
```